src/models/settings/settingsManager.ts

```diff
@@ -6,7 +6,7 @@ export interface ISettingsManager {
   getSettings: () => ISettings;
   getState: () => IState;
   setState: (state: IState) => void;
-  setStatus: (sts: ExtensionStatus) => void;
+  updateStatus: (sts: ExtensionStatus) => void;
```

Comment:
This interface is not consistent with the use of the function you make in the `SettingsManager` class. It should return an `IState`